### PR TITLE
[Model Refactor] Added linear_regression.py and a script for training LR models.

### PR DESCRIPTION
**Pull request by jakeval**
## Summary of changes
* Added a model training file for logistic regression models logistic_regression.py
* Added scripts/train_model.py to launch model training
* Added saved_models/ to save model results

## Feedback requested
The goal for this PR is:

* Code should be clean enough to make maintenance easy for the immediate lifespan of the project -- until results are finalized (1-2 months)
* Code should be clean enough that only minor refactors are required to achieve open-source quality
* Open-source quality is a nice-to-have but not a requirement

### CODE REVIEW

File: models/core/logistic_regression.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+        Saving is done with joblib.
+
+        Args:
+            model: The model to save.
+            model_dir: The directory to save the model under."""
+        super().save_model(model, model_dir)
+        joblib.dump(model, os.path.join(model_dir, MODEL_FILENAME))
+
+    def _load_model(self, model_dir: str) -> model_interface.Model:
```
Comment (bringingjoy) [inline models/core/logistic_regression.py:69]: This can have a one-line docstring given the simplicity of this function.

File: models/core/logistic_regression.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
+from sklearn import linear_model
+import pandas as pd
+import joblib
```
Comment (bringingjoy) [inline models/core/logistic_regression.py:3]: Why is this needed? Is it possible to simply load in the LogisticRegression model via sci-kit learn's interface?

File: models/core/logistic_regression.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        Args:
+            dataset: The training dataset to use.
+            dataset_info: Information on the training dataset.
+
+        Returns:
+            A trained Model."""
+        adapter = self._get_adapter(dataset, dataset_info)
+        lr = linear_model.LogisticRegression(**TRAINING_PARAMS)
+        dataset = adapter.transform(dataset)
+        X = dataset.drop(dataset_info.label_name, axis=1)
```
Comment (bringingjoy) [inline models/core/logistic_regression.py:52]: Better to use `training_data` and `labels` or other clearer names over `X` and `y`.

Comment (bringingjoy) [reply models/core/logistic_regression.py:52]: At this stage in the project, focus on consistency. This is an important style and readability concern for the future.

File: scripts/train_model.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import sys
```
Comment (bringingjoy) [inline scripts/train_model.py:1]: This script looks simple and straightforward to use.

**Review by bringingjoy — APPROVED** [review]
Overall, looks good. Keep in mind the lessons for the high-level abstractions that you added as part of this project for future software engineering and research code development.

File: models/core/logistic_regression.py (hunk 2 above)
Comment (jakeval) [reply models/core/logistic_regression.py:3]: [sklearn documentation](https://scikit-learn.org/stable/model_persistence.html) recommends joblib.
